engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -0,0 +1,142 @@
+require 'net/http'
+
+ANNOUNCE_URL = 'https://scc.suse.com/connect/subscriptions/systems'.freeze
+ACTIVATE_PRODUCT_URL = 'https://scc.suse.com/connect/systems/products'.freeze
+NET_HTTP_ERRORS = [
+  Errno::EINVAL,
+  Errno::ECONNRESET,
+  EOFError,
+  Net::HTTPBadResponse,
+  Net::HTTPForbidden,
+  Net::HTTPGone,
+  Net::HTTPBadRequest,
+  Net::HTTPFound,
+  Net::HTTPHeaderSyntaxError,
+  Net::ProtocolError,
+  Net::OpenTimeout,
+  Net::HTTPServerException,
+  Net::HTTPFatalError,
+  OpenSSL::SSL::SSLError,
+  Errno::EHOSTUNREACH,
+  Net::HTTPRetriableError
+].freeze
+
+module SccProxy
+  class << self
+
+    def headers(auth)
+      {
+        'accept' => 'application/json,application/vnd.scc.suse.com.v4+json',
+        'Content-Type' => 'application/json',
+        'Authorization' => auth
+      }
+    end
+
+    def prepare_scc_announce_request(uri_path, auth, params)
+      scc_request = Net::HTTP::Post.new(uri_path, headers(auth))
+      hw_info_keys = %i[cpus sockets hypervisor arch uuid cloud_provider]
+      hw_info = params['hwinfo'].symbolize_keys.slice(*hw_info_keys)
+      scc_request.body = {
+        hostname: params['hostname'],
+        hwinfo: hw_info
+      }.to_json
+      scc_request
+    end
+
+    def prepare_scc_request(uri_path, product, auth, token, email)
+      scc_request = Net::HTTP::Post.new(uri_path, headers(auth))
+      scc_request.body = {
+        token: token,
+        identifier: product.identifier,
+        version: product.version,
+        arch: product.arch,
+        release_type: product.release_type,
+        email: email || nil
+      }.to_json
+      scc_request
+    end
+
+    def announce_system_scc(auth, params)
+      uri = URI.parse(ANNOUNCE_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      scc_request = prepare_scc_announce_request(uri.path, auth, params)
+      response = http.request(scc_request)
+      response.error! unless response.code_type == Net::HTTPCreated
+
+      JSON.parse(response.body)
+    end
+
+    def scc_activate_product(product, auth, token, email, logger)
+      uri = URI.parse(ACTIVATE_PRODUCT_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      scc_request = prepare_scc_request(uri.path, product, auth, token, email)
+      response = http.request(scc_request)
+      logger.info(
+        "Response code is #{response.code} for the attempt to " \
+          "activate product #{product.product_string} to SCC"
+      )
+      return false unless response.code_type == Net::HTTPCreated
```

Comment:
I think it would be helpful to add detailed logging of the response in case the result is not successful.